[ES3] Consolidate tech preview limitations + serverless differences, … (#194)
elastic/docs-content: serverless/pages/serverless-differences.asciidoc

[[elasticsearch-differences]]
= Differences from other {es} offerings
++++
<titleabbrev>Serverless differences</titleabbrev>
++++

// :description: Understand how {es-serverless} differs from Elastic Cloud Hosted and self-managed offerings.
// :keywords: serverless, elasticsearch

> <<what-is-elasticsearch-serverless,{es-serverless}>> abstracts away infrastructure management from users, providing a fully managed {es} service.

Author: 💯

If you've used {es} before, you'll notice some differences in how you work with the service on serverless, because a number of APIs and settings are not required for serverless projects.

This guide helps you understand what's different, what's available, and how to work effectively when running {es} on {serverless-full}.

[discrete]
[[elasticsearch-differences-serverless-infrastructure-management]]
== Fully managed infrastructure

{es-serverless} manages all infrastructure automatically, including:

* Cluster scaling and optimization
* Node management and allocation
* Shard distribution and replication
* Resource utilization and monitoring

This fully managed approach means many traditional {es} infrastructure APIs and settings are not available to end users, as detailed in the following sections.

[discrete]
[[elasticsearch-differences-serverless-aps-settings]]
== APIs and settings

Because {es-serverless} manages infrastructure automatically, certain APIs and settings are not available, while others remain fully accessible.

[TIP]
====

> Refer to the {es-serverless} https://www.elastic.co/docs/api/[API Reference] for a complete list of available APIs.

Author: yes updated your suggestion with better link 👍

====

[discrete] 
[[elasticsearch-differences-serverless-api-availability]]
=== API availability

> The following categories of operations are *unavailable* since they are managed by the serverless platform:

Author: 200%

Infrastructure operations::
* All `_nodes/*` operations 
* All `_cluster/*` operations
* Most `_cat/*` operations, except for index-related operations such as `/_cat/indices` and `/_cat/aliases`

Storage and backup::
* All `_snapshot/*` operations
* Repository management operations

Index management:: 
* `indices/close` operations
* `indices/open` operations
* Recovery and stats operations
* Force merge operations

[discrete]
[[elasticsearch-differences-serverless-settings-availability]]  
=== Settings availability

In {es-serverless}, you can only configure index-level settings.
Cluster-level settings and node-level settings are not required by end users and the `elasticsearch.yml` file is fully managed by Elastic.

Available settings::

> * *Index-level settings*. Settings that control how {es} documents are processed, stored, and searched are available to end users, including:

Author: ```suggestion
* {ref}/index-modules.html#index-modules-settings[*Index-level settings*] control how {es} documents are processed, stored, and searched are available to end users, including:
```

** Analysis configuration
** Mapping parameters
** Search/query settings
** Indexing settings such as `refresh_interval`

Managed settings::

> * *Infrastructure-related settings*. Settings that affect cluster resources or data distribution are not available to end users, including:

Author: ```suggestion
*Infrastructure-related settings*: Settings that affect cluster resources or data distribution are not available to end users, including:
```

** Node configurations
** Cluster topology
** Shard allocation
** Resource management

[discrete]
[[elasticsearch-differences-serverless-api-errors]]
=== API error handling

> When attempting to use an unavailable API or setting, you'll receive a clear error message:

Reviewer: I'd remove `setting` from this (expect there is a different behavior for invalid settings). 

consider parenting this to `API availability` and renaming the section to `APIs` - that way, there's a clear pairing of "here's what's available, and here's what happens when you try to use something unavailable", instead of the settings stuff breaking it up

> When attempting to use an unavailable API or setting, you'll receive a clear error message:

Author: ```suggestion
When attempting to use an unavailable API, you'll receive a clear error message:
```

[source,json]
----
{
 "error": {
   "root_cause": [
     {
       "type": "api_not_available_exception",
       "reason": "Request for uri [/<API_ENDPOINT>] with method [<METHOD>] exists but is not available when running in serverless mode"
     }
   ],
   "status": 410
 }
}
----

[discrete]
[[elasticsearch-differences-serverless-feature-categories]]
== Feature availability

Some features that are available in Elastic Cloud Hosted and self-managed offerings are not available in {es-serverless}.

> These features have either been replaced by a new feature, or are not applicable in the new Serverless architecture:

Reviewer: this seems to be the only capitalized `Serverless` here - could lowercase for consistency

Reviewer: I think these "unavailable due to a better replacement" features need their own category

> These features have either been replaced by a new feature, or are not applicable in the new Serverless architecture:

Author: ```suggestion
These features have either been replaced by a new feature, or are not applicable in the new serverless architecture:
```

     }
   ],
   "status": 410
 }
}
----

[discrete]
[[elasticsearch-differences-serverless-settings-availability]]  
== Settings availability

In {es-serverless}, you can only configure {ref}/index-modules.html#index-modules-settings[index-level settings].
Cluster-level settings and node-level settings are not required by end users and the `elasticsearch.yml` file is fully managed by Elastic.

Available settings::
*Index-level settings*: Settings that control how {es} documents are processed, stored, and searched are available to end users. These include:
* Analysis configuration
* Mapping parameters
* Search/query settings
* Indexing settings such as `refresh_interval`

Managed settings::
*Infrastructure-related settings*: Settings that affect cluster resources or data distribution are not available to end users. These include:
* Node configurations
* Cluster topology
* Shard allocation
* Resource management

[discrete]
[[elasticsearch-differences-serverless-feature-categories]]
== Feature availability

Some features that are available in Elastic Cloud Hosted and self-managed offerings are not available in {es-serverless}.

> These features have either been replaced by a new feature, or are not applicable in the new serverless architecture.

Reviewer: this does not cover planned features

> These features have either been replaced by a new feature, or are not applicable in the new serverless architecture.

Reviewer: ```suggestion
These features have either been replaced by a new feature, are planned to be released in future, or are not applicable in the new serverless architecture.
```

+
Kibana Alerts allows rich integrations across use cases like APM, metrics, security, and uptime. Prepackaged rule types simplify setup and
hide the details of complex, domain-specific detections, while providing a consistent interface across Kibana.

[discrete]
[[elasticsearch-differences-serverless-feature-planned]]
=== Planned features

The following features are planned for future support in all {serverless-full} projects:

* Reindexing from remote clusters 
* Cross-project search and replication
* Snapshot and restore
* Migrations from non-serverless deployments
* Audit logging
* Authentication realms (native/SAML/OIDC/Kerberos/JWT)
* Clone index API
* Traffic filtering and VPCs

> The following _{es-serverless}_ project-specific features are planned for future support:

Reviewer: I don't think these italics help

Author: 375%

> The following _{es-serverless}_ project-specific features are planned for future support:

Author: ```suggestion
The following {es-serverless} project-specific features are planned for future support:
```

* {ref}/behavioral-analytics-overview.html[Behavioral Analytics]
* {ref}/search-application-overview.html[Search Applications]
* {enterprise-search-ref}/crawler.html[Managed web crawler]

> ** You can use the https://github.com/elastic/crawler[self-managed web crawler] in the meantime.

Reviewer: prefer child paragraph rather than a child bullet

* {ref}/es-native-connectors.html[Elastic managed Search connectors]

> ** You can use the {ref}/es-build-connector.html[self-managed Search connectors] in the meantime.

Reviewer: should also not be in a child bullet

also: 
```suggestion
** You can use {ref}/es-build-connector.html[self-managed Search connectors] in the meantime.
```
